### PR TITLE
Take and return new RMW_DURATION_INFINITE correctly

### PR DESCRIPTION
**Pull request by emersonknapp**
Depends on https://github.com/ros2/rmw/pull/301

Meets the new API expectations introduced in https://github.com/ros2/rmw/pull/301 - to accept and return `RMW_DURATION_INFINITE` for cross-implementation consistency.

### CODE REVIEW

**Comment by emersonknapp** [comment]
@asorbini this is probably not worth merging here, right, and should instead go to `rmw_connextdds`?

**Comment by asorbini** [comment]
@emersonknapp I can confirm I'd like support for this special value to be added to `rmw_connextdds`, but I can't really comment on whether it shouldn't be merged to `rmw_connext_cpp` too. I'm not sure what the plans are for EOL'ing the package, beside phasing it out of the source tree. I'm assuming development will eventually stop, but a maintainer should comment on that.

Feel free to open a PR for `rmw_connextdds` though, I'd be happy to review/merge.

Keep in mind that @ivanpauno is also working on some changes related to QoS configuration (see [rticommunity/rmw_connextdds#7](https://github.com/rticommunity/rmw_connextdds/pull/7)), so you might want to coordinate these additions with him.

**Comment by clalancette** [comment]
My personal view on this is that the work here is done, so we should just go ahead and merge this.  I wouldn't spend much *additional* time here, given that it is on its way out, but "on its way out" is not the same thing as "gone".

**Comment by emersonknapp** [comment]
Gist: https://gist.githubusercontent.com/emersonknapp/08a0496aa3726c8053285e6def25ab07/raw/1550d9ed37d12a2824dfb876996e6b09bf6844ba/ros2.repos
BUILD args: --packages-up-to rmw_connext_cpp rcl
TEST args: --packages-select rmw_connext_cpp rcl
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/7857
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13917/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8765/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11623/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14005/)

**Comment by emersonknapp** [comment]
^ the `rmw_connext` packages are being ignored by the build, I guess we've already disabled connext in CI?

**Comment by asorbini** [comment]
> ^ the `rmw_connext` packages are being ignored by the build, I guess we've already disabled connext in CI?

My changes have only made it into a [branch of `ros2/ci`](https://github.com/ros2/ci/tree/asorbini/rmw_connextdds).

I think `rmw_connext_cpp` was already disabled (as in, completely ignored) for `linux-aarch64` and `armhf`. See [this change from my commit](https://github.com/ros2/ci/commit/2ebe8e138c0661b6145f41f1be94b22f273cf775#diff-254dbe084a5da49fd042f08f46f9d2633cc27079514084b1e7f0152ddb44bf36L139) for reference.

**Comment by emersonknapp** [comment]
Oh! You're right it's only on ARM. Never mind then, we can just ignore that build

**Comment by clalancette** [comment]
@ros-pull-request-builder retest this please

**Comment by emersonknapp** [comment]
@ahcorde @mjcarroll as maintainers do you have any input on this?

**Comment by emersonknapp** [comment]
@clalancette friendly ping - should we merge this?

File: rmw_connext_shared_cpp/src/qos.cpp
Comment (clalancette) [inline rmw_connext_shared_cpp/src/qos.cpp:32]: This is a dead line of code now, is it not?

**Review by clalancette — CHANGES_REQUESTED** [review]
Sorry, I just noticed one issue with this PR.  After that, I'm happy to approve.

File: rmw_connext_shared_cpp/src/qos.cpp
Comment (emersonknapp) [reply rmw_connext_shared_cpp/src/qos.cpp:32]: good catch!

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI (which is probably redundant, but I'm being a little more conservative as we approach the Galactic deadline).

**Comment by emersonknapp** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14042/)
* Linux-aarch64 (not supported by rmw_connext)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11735/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14133/)

**Pull request merged**
